packages/plugins/graphql-plugin-auth/src/Neo4jGraphQLAuthJWTPlugin.test.ts

```diff
@@ -39,6 +39,30 @@ describe("Neo4jGraphQLAuthJWTPlugin", () => {
         expect(decoded).toMatchObject(payload);
     });
 
+    test("should decode token when secret is a generic", async () => {
```

Comment:
Thanks for the test case!
Would you be able to add another test case if the `secret` function returns `null` or `undefined`?
Similar to:
```javascript
secret: () => {
    return null;
},
```

Reply:
Hi @tbwiss ,

about the `null` test case I'm not sure. 
In two cases the secret can be null.
1. The `tryToResolveKeys` is not called properly in `wrapper.ts`
2. The getter for secret is returning null.

in both cases the `decode` will return `undefined`. We can create a test for the first case to check if `debug()` is called in case the `tryToResolveKeys` is not called but the second case I think would be test case for the host application.

for example if I assign a function to `secret` from my service, I have to unit-test that function inside that service right? so it doesn't return null.

but maybe in both cases we can add a test which says if `secret` was null then check if debug is called but as I mentioned earlier the `debug()` will never be called because apparently the `debug.enabled` is always false and we are not setting it to true anywhere. 

let me know if we need to change this, then I will try to add a `Debug.Debugger` to the inputs so then we can maybe mock it

Reply:
Yes, I understand your point!
We can't however rely on the user/customer/service code to be unit tested. Therefore we have to expect `null` as a potential value of `secret()`.
If it's possible for you to add a test for this scenario that would be really good. Once that is in place I would say we can merge this PR.

Reply:
on it 👍

Reply:
Hi @tbwiss 

I added a test but it is not really the case where the computed value is null, that value can't be null since the function must return only type `Secret`.

![image](https://user-images.githubusercontent.com/57224511/212570914-6a9cb86e-882f-4ffd-ace7-b506cb448d2e.png)

but I now check if the `secret` is passed as a function and if the result value is still null, it throws an exception.

Reply:
@farhadnowzari, thank you!